### ISSUE TITLE
[vcd_org_user] updating deployed_vm_quota and stored_vm_quota value to 0 doesn't work

### ISSUE THREAD

**Issue by c35sys** [vmware/terraform-provider-vcd#799]
Hello,

When updating **deployed_vm_quota** and **stored_vm_quota** values on an existing user to **0**, it doesn't work.
It is working with another value, such as 2000 for example, increasing and decreasing.
It works **only** when creating the user with an initial 0 value for both arguments.

### Terraform Version
Terraform v1.1.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/vault v3.0.1
+ provider registry.terraform.io/vmware/vcd v3.5.1

### Affected Resource(s)
Please list the resources as a list, for example:
- vcd_org_user

### Terraform Configuration Files

```hcl
resource "vcd_org_user" "org_vdc_admin" {
  org = var.my_org

  name              = "admin"
  description       = "${var.my_org} org admin"
  role              = "Organization Administrator"
  password          = data.vault_generic_secret.vcd_org_admin_account.data["value"]
  deployed_vm_quota = 0
  stored_vm_quota   = 0

  depends_on = [
    vcd_org.org,
  ]

}
```

### Debug Output
```shell

-----------------------------------------------------: timestamp=2022-03-03T10:36:05.570+0100
2022-03-03T10:36:05.571+0100 [INFO]  provider.terraform-provider-vault_v3.0.1_x4: 2022/03/03 10:36:05 [INFO] Using Vault token with the following policies: default, infra, sandbox: timestamp=2022-03-03T10:36:05.570+0100
2022-03-03T10:36:05.571+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-03-03T10:36:05.572+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/vault/3.0.1/linux_amd64/terraform-provider-vault_v3.0.1_x4 pid=15824
2022-03-03T10:36:05.572+0100 [DEBUG] provider: plugin exited
2022-03-03T10:36:05.572+0100 [TRACE] vertex "provider[\"registry.terraform.io/hashicorp/vault\"] (close)": visit complete
2022-03-03T10:36:06.399+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Called downstream: tf_provider_addr=provider tf_req_id=fd852bce-d9b0-d270-b283-4e11c6eb7a2b tf_rpc=Configure @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:392 tf_proto_version=5 @module=sdk.proto timestamp=2022-03-03T10:36:06.399+0100
2022-03-03T10:36:06.399+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Served request: @module=sdk.proto tf_proto_version=5 tf_req_id=fd852bce-d9b0-d270-b283-4e11c6eb7a2b @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:398 tf_provider_addr=provider tf_rpc=Configure timestamp=2022-03-03T10:36:06.399+0100
2022-03-03T10:36:06.399+0100 [TRACE] vertex "provider[\"registry.terraform.io/vmware/vcd\"]": visit complete
2022-03-03T10:36:06.399+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin (expand)": starting visit (*terraform.nodeExpandApplyableResource)
2022-03-03T10:36:06.399+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin (expand)": expanding dynamic subgraph
2022-03-03T10:36:06.399+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin (expand)": entering dynamic subgraph
2022-03-03T10:36:06.399+0100 [TRACE] vertex "vcd_org.org (expand)": starting visit (*terraform.nodeExpandApplyableResource)
2022-03-03T10:36:06.399+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin": starting visit (*terraform.NodeApplyableResource)
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org.org (expand)": expanding dynamic subgraph
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin": visit complete
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin (expand)": dynamic subgraph completed successfully
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin (expand)": visit complete
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org.org (expand)": entering dynamic subgraph
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org.org": starting visit (*terraform.NodeApplyableResource)
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org.org": visit complete
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org.org (expand)": dynamic subgraph completed successfully
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org.org (expand)": visit complete
2022-03-03T10:36:06.400+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin": starting visit (*terraform.NodeApplyableResourceInstance)
2022-03-03T10:36:06.400+0100 [TRACE] readDiff: Read Update change from plan for vcd_org_user.org_vdc_admin
2022-03-03T10:36:06.400+0100 [TRACE] readResourceInstanceState: reading state for vcd_org_user.org_vdc_admin
2022-03-03T10:36:06.400+0100 [TRACE] upgradeResourceState: schema version of vcd_org_user.org_vdc_admin is still 0; calling provider "vcd" for any other minor fixups
2022-03-03T10:36:06.400+0100 [TRACE] GRPCProvider: UpgradeResourceState
2022-03-03T10:36:06.400+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Received request: tf_resource_type=vcd_org_user tf_rpc=UpgradeResourceState @module=sdk.proto tf_proto_version=5 @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:520 tf_provider_addr=provider tf_req_id=49d2770c-0662-c376-559c-637f5ef9d99e timestamp=2022-03-03T10:36:06.400+0100
2022-03-03T10:36:06.400+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Calling downstream: tf_proto_version=5 tf_req_id=49d2770c-0662-c376-559c-637f5ef9d99e tf_resource_type=vcd_org_user @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:527 @module=sdk.proto tf_provider_addr=provider tf_rpc=UpgradeResourceState timestamp=2022-03-03T10:36:06.400+0100
2022-03-03T10:36:06.401+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Called downstream: @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:533 @module=sdk.proto tf_proto_version=5 tf_provider_addr=provider tf_req_id=49d2770c-0662-c376-559c-637f5ef9d99e tf_resource_type=vcd_org_user tf_rpc=UpgradeResourceState timestamp=2022-03-03T10:36:06.401+0100
2022-03-03T10:36:06.401+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Served request: tf_resource_type=vcd_org_user @module=sdk.proto tf_proto_version=5 tf_provider_addr=provider tf_rpc=UpgradeResourceState @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:539 tf_req_id=49d2770c-0662-c376-559c-637f5ef9d99e timestamp=2022-03-03T10:36:06.401+0100
2022-03-03T10:36:06.401+0100 [TRACE] readDiff: Read Update change from plan for vcd_org_user.org_vdc_admin
2022-03-03T10:36:06.402+0100 [TRACE] Re-validating config for "vcd_org_user.org_vdc_admin"
2022-03-03T10:36:06.402+0100 [TRACE] GRPCProvider: ValidateResourceConfig
2022-03-03T10:36:06.402+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Received request: tf_rpc=ValidateResourceTypeConfig @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:495 tf_proto_version=5 tf_provider_addr=provider tf_req_id=21365549-40fd-0276-35f1-2f922100b431 tf_resource_type=vcd_org_user @module=sdk.proto timestamp=2022-03-03T10:36:06.402+0100
2022-03-03T10:36:06.402+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Calling downstream: tf_provider_addr=provider tf_req_id=21365549-40fd-0276-35f1-2f922100b431 tf_resource_type=vcd_org_user tf_rpc=ValidateResourceTypeConfig tf_proto_version=5 @module=sdk.proto @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:502 timestamp=2022-03-03T10:36:06.402+0100
2022-03-03T10:36:06.402+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Called downstream: @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:508 tf_resource_type=vcd_org_user @module=sdk.proto tf_proto_version=5 tf_provider_addr=provider tf_req_id=21365549-40fd-0276-35f1-2f922100b431 tf_rpc=ValidateResourceTypeConfig timestamp=2022-03-03T10:36:06.402+0100
2022-03-03T10:36:06.402+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Served request: @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:514 tf_proto_version=5 tf_req_id=21365549-40fd-0276-35f1-2f922100b431 tf_resource_type=vcd_org_user tf_rpc=ValidateResourceTypeConfig @module=sdk.proto tf_provider_addr=provider timestamp=2022-03-03T10:36:06.402+0100
2022-03-03T10:36:06.403+0100 [TRACE] GRPCProvider: PlanResourceChange
2022-03-03T10:36:06.403+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Received request: tf_resource_type=vcd_org_user @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:570 @module=sdk.proto tf_proto_version=5 tf_provider_addr=provider tf_req_id=587dceb1-b183-c591-07fd-5ba5e3ce808e tf_rpc=PlanResourceChange timestamp=2022-03-03T10:36:06.403+0100
2022-03-03T10:36:06.403+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Calling downstream: @module=sdk.proto tf_proto_version=5 tf_resource_type=vcd_org_user tf_rpc=PlanResourceChange tf_provider_addr=provider tf_req_id=587dceb1-b183-c591-07fd-5ba5e3ce808e @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:577 timestamp=2022-03-03T10:36:06.403+0100
2022-03-03T10:36:06.404+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Called downstream: tf_provider_addr=provider tf_resource_type=vcd_org_user tf_rpc=PlanResourceChange @module=sdk.proto tf_req_id=587dceb1-b183-c591-07fd-5ba5e3ce808e @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:583 tf_proto_version=5 timestamp=2022-03-03T10:36:06.404+0100
2022-03-03T10:36:06.404+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Served request: @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:589 @module=sdk.proto tf_proto_version=5 tf_req_id=587dceb1-b183-c591-07fd-5ba5e3ce808e tf_resource_type=vcd_org_user tf_provider_addr=provider tf_rpc=PlanResourceChange timestamp=2022-03-03T10:36:06.404+0100
2022-03-03T10:36:06.404+0100 [WARN]  Provider "registry.terraform.io/vmware/vcd" produced an invalid plan for vcd_org_user.org_vdc_admin, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .email_address: planned value cty.StringVal("") for a non-computed attribute
      - .enabled: planned value cty.True for a non-computed attribute
      - .is_locked: planned value cty.False for a non-computed attribute
      - .provider_type: planned value cty.StringVal("INTEGRATED") for a non-computed attribute
      - .telephone: planned value cty.StringVal("") for a non-computed attribute
      - .full_name: planned value cty.StringVal("") for a non-computed attribute
      - .instant_messaging: planned value cty.StringVal("") for a non-computed attribute
      - .is_group_role: planned value cty.False for a non-computed attribute
      - .take_ownership: planned value cty.False for a non-computed attribute
2022-03-03T10:36:06.404+0100 [TRACE] checkPlannedChange: Verifying that actual change (action Update) matches planned change (action Update)
2022-03-03T10:36:06.404+0100 [TRACE] readResourceInstanceState: reading state for vcd_org_user.org_vdc_admin
2022-03-03T10:36:06.404+0100 [TRACE] upgradeResourceState: schema version of vcd_org_user.org_vdc_admin is still 0; calling provider "vcd" for any other minor fixups
2022-03-03T10:36:06.404+0100 [TRACE] GRPCProvider: UpgradeResourceState
2022-03-03T10:36:06.405+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Received request: tf_req_id=0d14d08f-47e1-939d-18c1-f026dda91881 tf_resource_type=vcd_org_user tf_rpc=UpgradeResourceState @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:520 tf_proto_version=5 @module=sdk.proto tf_provider_addr=provider timestamp=2022-03-03T10:36:06.405+0100
2022-03-03T10:36:06.405+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Calling downstream: tf_req_id=0d14d08f-47e1-939d-18c1-f026dda91881 tf_rpc=UpgradeResourceState tf_proto_version=5 tf_provider_addr=provider @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:527 @module=sdk.proto tf_resource_type=vcd_org_user timestamp=2022-03-03T10:36:06.405+0100
2022-03-03T10:36:06.405+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Called downstream: @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:533 tf_rpc=UpgradeResourceState @module=sdk.proto tf_proto_version=5 tf_provider_addr=provider tf_req_id=0d14d08f-47e1-939d-18c1-f026dda91881 tf_resource_type=vcd_org_user timestamp=2022-03-03T10:36:06.405+0100
2022-03-03T10:36:06.405+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Served request: tf_req_id=0d14d08f-47e1-939d-18c1-f026dda91881 tf_resource_type=vcd_org_user @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:539 @module=sdk.proto tf_proto_version=5 tf_provider_addr=provider tf_rpc=UpgradeResourceState timestamp=2022-03-03T10:36:06.405+0100
2022-03-03T10:36:06.405+0100 [INFO]  Starting apply for vcd_org_user.org_vdc_admin
2022-03-03T10:36:06.406+0100 [DEBUG] vcd_org_user.org_vdc_admin: applying the planned Update change
2022-03-03T10:36:06.406+0100 [TRACE] GRPCProvider: ApplyResourceChange
2022-03-03T10:36:06.406+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Received request: tf_proto_version=5 tf_provider_addr=provider tf_req_id=01ae63b5-7249-dfda-c146-5fcdf3d5a2f8 tf_resource_type=vcd_org_user @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:595 tf_rpc=ApplyResourceChange @module=sdk.proto timestamp=2022-03-03T10:36:06.406+0100
2022-03-03T10:36:06.406+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Calling downstream: tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:602 @module=sdk.proto tf_req_id=01ae63b5-7249-dfda-c146-5fcdf3d5a2f8 tf_proto_version=5 tf_provider_addr=provider tf_resource_type=vcd_org_user timestamp=2022-03-03T10:36:06.406+0100
2022-03-03T10:36:07.339+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Called downstream: tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:608 tf_provider_addr=provider tf_req_id=01ae63b5-7249-dfda-c146-5fcdf3d5a2f8 tf_resource_type=vcd_org_user @module=sdk.proto tf_proto_version=5 timestamp=2022-03-03T10:36:07.339+0100
2022-03-03T10:36:07.339+0100 [TRACE] provider.terraform-provider-vcd_v3.5.1: Served request: @module=sdk.proto tf_provider_addr=provider tf_req_id=01ae63b5-7249-dfda-c146-5fcdf3d5a2f8 @caller=github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:614 tf_resource_type=vcd_org_user tf_rpc=ApplyResourceChange tf_proto_version=5 timestamp=2022-03-03T10:36:07.339+0100
2022-03-03T10:36:07.339+0100 [WARN]  Provider "provider[\"registry.terraform.io/vmware/vcd\"]" produced an unexpected new value for vcd_org_user.org_vdc_admin, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .deployed_vm_quota: was cty.NumberIntVal(0), but now cty.NumberIntVal(1000)
      - .stored_vm_quota: was cty.NumberIntVal(0), but now cty.NumberIntVal(1000)
2022-03-03T10:36:07.339+0100 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for vcd_org_user.org_vdc_admin
2022-03-03T10:36:07.339+0100 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: writing state object for vcd_org_user.org_vdc_admin
2022-03-03T10:36:07.339+0100 [TRACE] evalApplyProvisioners: vcd_org_user.org_vdc_admin is not freshly-created, so no provisioning is required
2022-03-03T10:36:07.339+0100 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for vcd_org_user.org_vdc_admin
2022-03-03T10:36:07.339+0100 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: writing state object for vcd_org_user.org_vdc_admin
2022-03-03T10:36:07.340+0100 [TRACE] vertex "vcd_org_user.org_vdc_admin": visit complete
2022-03-03T10:36:07.341+0100 [TRACE] vertex "provider[\"registry.terraform.io/vmware/vcd\"] (close)": starting visit (*terraform.graphNodeCloseProvider)
2022-03-03T10:36:07.341+0100 [TRACE] GRPCProvider: Close
2022-03-03T10:36:07.341+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2022-03-03T10:36:07.342+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/vmware/vcd/3.5.1/linux_amd64/terraform-provider-vcd_v3.5.1 pid=15833
2022-03-03T10:36:07.342+0100 [DEBUG] provider: plugin exited
2022-03-03T10:36:07.342+0100 [TRACE] vertex "provider[\"registry.terraform.io/vmware/vcd\"] (close)": visit complete
2022-03-03T10:36:07.342+0100 [TRACE] vertex "root": starting visit (*terraform.nodeCloseModule)
2022-03-03T10:36:07.342+0100 [TRACE] vertex "root": visit complete

```
### Expected Behavior

The values should be set to 0.

### Actual Behavior
Terraform says it is working, but it doesn't do anything.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create a vcd_org_user
2. Set deployed_vm_quota** and **stored_vm_quota** to 200
3. `terraform apply`
4. Set deployed_vm_quota** and **stored_vm_quota** to 0
5. `terraform apply`

**Comment by mikeletux** [comment]
Hello @c35sys , let's me try to reproduce the issue and see how we can fix this. Thanks for the feedback! I'll be try to come back to you as soon as possible!

**Comment by mikeletux** [comment]
@c35sys just wanted to confirm you that I've been able to reproduce the issue. I'll try to find some time to fix it!
Thanks for reporting this bug!

**Comment by c35sys** [comment]
Thanks for the update @mikeletux!

**Comment by mikeletux** [comment]
@c35sys yesterday I had some time to look into this and I spotted the issue. It is a bug on go-vcd SDK.
I already opened a PR for my colleagues to check and if they agree, merge. 

You can check it here https://github.com/vmware/go-vcloud-director/pull/445.

Thanks for giving feedback and helping us! This helps us building a better provider for everyone to use! 🚀 🚀 🚀

**Comment by mikeletux** [comment]
@c35sys I've merged the PR that solves this issue into main in https://github.com/vmware/go-vcloud-director repo.
This means the issue will be solved when v3.6.0 is released!

If you want to test it before the release, you can build this provider using the main branch for go-vcloud-director. 
Let me know also if we can close this issue!

**Comment by c35sys** [comment]
Thanks @mikeletux, I'll try as soon as possible !
